src/Compilers/Core/Portable/Syntax/SyntaxNodeOrTokenList.cs

```diff
@@ -316,10 +316,10 @@ private static SyntaxNodeOrTokenList CreateList(List<SyntaxNodeOrToken> items)
                 return default(SyntaxNodeOrTokenList);
             }
 
-            var newGreen = GreenNode.CreateList(items.Select(n => n.UnderlyingNode!))!;
+            var newGreen = GreenNode.CreateList(items, n => n.UnderlyingNode!)!;
             if (newGreen.IsToken)
             {
-                newGreen = GreenNode.CreateList(new[] { newGreen }, alwaysCreateListNode: true)!;
+                newGreen = Syntax.InternalSyntax.SyntaxList.List(new[] { newGreen });
```

Comment:
Why this change? Unless this is a particular hotspot, I'd prefer just passing an identity selector.

Reply:
I prefered this because it allows me to remove the optional bool parameter. Should I revert this?

Reply:
i'm good with this.